main.py

```diff
@@ -0,0 +1,63 @@
+import config
+from CanvasAPI import *
+
+
+def getCanvasInstance():
+    return CanvasAPI(config.Canvas.API_BASE_URL, authZToken=config.Canvas.API_AUTHZ_TOKEN)
+
+
+def getCourseIDsWithOutcome(canvas, courseIDs, outcome):
+    matchingCourseIDs = set()
+    for courseID in courseIDs:
+        courseOutcomeGroupLinks = canvas.getCoursesOutcomeGroupLinksObjects(courseID)
+
+        # Is it possible to short-circuit this using itertools?
+        matchingCourseIDs.update(
+            {courseID for outcomeLink in courseOutcomeGroupLinks
+             if outcomeLink.outcome.id == outcome.id}
+        )
+    return matchingCourseIDs
+
+
+def getCourseAssignmentsWithOutcome(canvas, courseIDs, outcome):
```

Comment:
Yes, I wrestled with this one for a good amount of time one day.  I'm sure it can be simplified (or at least Pythonicated [I think I coined a new word!]) more like the previous method, `getCourseIDsWithOutcome`.  The problem I had, though is that by using `break`, I could stop the loop sooner than I could with list comprehensions, map, filter, etc.

I think I did pretty well with that in `getCourseIDsWithOutcome`, but I think I could a little better there, too.  Of course, I had the holy grail in mind with both of these: Code that's simple, elegant, maintainable, and efficient.  Sometimes it's tough to get all of those together in the same place.